pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -90,9 +89,20 @@ public List<String> assignSegment(String segmentName, Map<String, Map<String, St
     InstancePartitions instancePartitions = typeToInstancePartitions.getValue();
     _logger.info("Assigning segment: {} with instance partitions: {} for table: {}", segmentName, instancePartitions,
         _tableNameWithType);
+
+    SegmentAssignmentStrategy segmentAssignmentStrategy = null;
```

Comment:
I think that the L93-105 can be written as the following:

```
    List<String> instancesAssigned;
    if (instancePartitionsType == InstancePartitionsType.COMPLETED) {
      // Gets Segment assignment strategy for instance partitions
      SegmentAssignmentStrategy segmentAssignmentStrategy = SegmentAssignmentStrategyFactory
          .getSegmentAssignmentStrategy(_helixManager, _tableConfig, instancePartitions, instancePartitionsType);
      instancesAssigned = segmentAssignmentStrategy.assignSegment(segmentName,
                currentAssignment, instancePartitions, InstancePartitionsType.COMPLETED);
    } else {
      instancesAssigned = assignConsumingSegment(SegmentName, instancePartitions);
    }
```
```

Reply:
updated